### PR TITLE
[RQ-802] feat: integration of session replay with test this rule

### PR DESCRIPTION
**Pull request by nafees87n**
Closes issue:  RQ-802

## ✅ Checklist:

- [ ] Make sure linting and unit tests pass.
- [ ] No install/build warnings introduced.
- [ ] Verified UI in browser.
- [ ] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

**Review by rohanmathur91 — CHANGES_REQUESTED** [review]
UI is broken for large URLs
Screenshot 2023-09-18 at 4 27 59 PM

File: app/src/components/features/rules/TestThisRule/index.css [hunk 1]
```diff
@@ -88,6 +105,16 @@
   animation: highlight-report 4s ease-in-out;
 }
 
+.rq-modal.draft-session-modal {
+  width: 90vw !important;
```
Comment (rohanmathur91) [inline app/src/components/features/rules/TestThisRule/index.css:109]: you can pass `Width` prop directly to the <RQModal> component

**Review by rohanmathur91 — APPROVED** [review]

**Review by wrongsahil — APPROVED** [review]

**Review by wrongsahil — CHANGES_REQUESTED** [review]
We will need to add compatibilityFlag for starting recording with session enabled

File: app/src/components/features/rules/TestThisRule/index.tsx [hunk 2]
```diff
@@ -26,8 +28,10 @@ export const TestThisRuleRow: React.FC = () => {
   const isCurrentlySelectedRuleHasUnsavedChanges = useSelector(getIsCurrentlySelectedRuleHasUnsavedChanges);
   const currentlySelectedRuleData = useSelector(getCurrentlySelectedRuleData);
   const testThisRuleBoxRef = useRef(null);
+
   const [pageUrl, setPageUrl] = useState("");
   const [error, setError] = useState(null);
+  const [recordTestPage, setRecordTestPage] = useState<boolean>(!!isFeatureCompatible(FEATURES.TEST_THIS_RULE_SESSION));
```
Comment (wrongsahil) [inline app/src/components/features/rules/TestThisRule/index.tsx:34]: This is uneccessary as event if we send record=true to older extension, it wont still record the page.

**Review by wrongsahil — CHANGES_REQUESTED** [review]

**Review by wrongsahil — APPROVED** [review]

**Pull request merged**